@xen-orchestra/lite/src/components/Table.tsx

```diff
@@ -19,55 +54,320 @@ interface ParentState {}
+    onShowByChange,
+    page,
+    reverse,
+    rowPerPageOptions = [10, 25, 50, 100],
+    showByValue,
+  }) => (
+    <Box
+      sx={{
+        display: 'flex',
+        justifyContent: nbSelectedItems !== undefined ? 'space-between' : 'right',
+      }}
+    >
+      {nbSelectedItems !== undefined && (
+        <StyledNbSelectedItems reverse={reverse}>
+          <IntlMessage id='itemSelected' values={{ nSelected: nbSelectedItems }} />
+        </StyledNbSelectedItems>
+      )}
+      <StyledPaginationContainer>
+        <StyledPaginationFlexRow>
+          <StyledPaginationText>
+            {page * showByValue + 1}-{page * showByValue + nbItemsOnThePage} of {nbTotalItems}
+            {dataType !== undefined && ` ${dataType}`}
+          </StyledPaginationText>
+          <StyledNavButton color='primary' size='small' disabled={page === 0} onClick={() => onPaginationChange('', 0)}>
+            <SkipPreviousIcon />
+          </StyledNavButton>
+          <StyledNavButton
+            color='primary'
+            disabled={page === 0}
+            onClick={() => onPaginationChange('', page - 1)}
+            size='small'
+          >
+            <NavigateBeforeIcon />
+          </StyledNavButton>
+          <StyledNavButton
+            color='primary'
+            disabled={page >= nbTotalPage - 1}
+            onClick={() => onPaginationChange('', page + 1)}
+            size='small'
+          >
+            <NavigateNextIcon />
+          </StyledNavButton>
+          <StyledNavButton
+            color='primary'
+            disabled={page >= nbTotalPage - 1}
+            onClick={() => onPaginationChange('', nbTotalPage - 1)}
+            size='small'
+          >
+            <SkipNextIcon />
+          </StyledNavButton>
+        </StyledPaginationFlexRow>
+        <StyledPaginationFlexRow>
+          <StyledPaginationText>Show by</StyledPaginationText>
+          <Select
+            color='primary'
+            noNone
+            onChange={onShowByChange}
+            optionRenderer={item => item}
+            options={rowPerPageOptions}
+            required
+            sx={{
+              backgroundColor: '#E8E8E8',
+              border: 'none',
+              color: '#007bff',
+              height: '24px',
+            }}
+            value={showByValue}
+            valueRenderer={item => item}
+          />
+        </StyledPaginationFlexRow>
+      </StyledPaginationContainer>
+    </Box>
+  )
+)
+
+const Table = withState<State, Props, Effects, Computed, ParentState, ParentEffects>(
+  {
+    effects: {
+      handlePaginationChange: function (_, page) {
+        const reg = new RegExp(this.props.stateUrlParam + '_page=\\d+')
+        let searchParams = this.props.location.search
+        if (reg.test(searchParams)) {
+          searchParams = searchParams.replace(reg, `${this.props.stateUrlParam}_page=${page}`)
+        } else {
+          searchParams += `?${this.props.stateUrlParam}_page=${page}`
```

Comment:
If you already have some search params, this would need to be a `&` instead of a `?`.